### PR TITLE
Improve "What's new" feature visibility

### LINKED ISSUE #20283: Improve our "What's new" feature to increase visibility
When there is a new "what's new" entry we already show a slightly different icon in the top menu. However, it's barely visible.

Maybe we could make this stand out a bit more eg by directly opening the "What's new" popup when there is a new entry and show the information and people can click it away? This worked really well for us last time on the Cloud and guaranteed everyone will notice it. Or there could be also other solutions.

From what we're seeing so far not too many are likely clicking on it when there is actually a what's new entry. We aren't tracking this though so this might not be correct but likely since newly linked pages from there are barely opened.

### PR DESCRIPTION
**Pull request by bx80**
### Description:

*** Closed in favour of #21271 ***

Fixes #20283 

- Adds a badge overlay to the "What's new" top menu notification icon to show the number of new changes.
- Shows the "What's new" dialog as a popup automatically when logging in if there are new changes.

image

The badge is implemented as an extra parameter that can be provided to any `core/Menu/MenuAbstract` descendant, the topbar template now supports this option so any top menu item can show a badge. If required, later on we could extend badge support to other templates / components which display menu items.

For testing the What's New auto popup, once the popup is viewed then the changes will not be shown again, to make it a bit easier to test here is a quick SQL insert to add a new random change item via the database:
```
INSERT INTO matomo.matomo_changes 
(created_time, plugin_name, version, title, description, link_name, link)
VALUES (DATE_FORMAT(NOW(), '%Y-%m-%d %h:%i:%s'), 'Test Plugin', '1.2.3', 
CONCAT('Test change ', (FLOOR(1 + RAND() * 1000))), 'Test details', 'Here is a link', 'https://matomo.org/');
```

For testing, the `core/Changes/Model` is now partially mocked to prevent loading of real plugin changes which cause the popup to appear on UI screenshots. 

Setting the template variable `hideWhatIsNew = true` will prevent the "What's New" popup from being shown for that template, this is applied for the no data screen.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: core/Plugin/Controller.php [hunk 1]
```diff
@@ -309,6 +312,11 @@ protected function renderTemplateAs($template, array $variables = array(), $view
             $viewType = $this instanceof ControllerAdmin ? 'admin' : 'basic';
         }
 
+        // Set early so it can available for setGeneralVariables method calls
```
Comment (sgiehl) [inline core/Plugin/Controller.php]:
```suggestion
        // Set early so it is available for setGeneralVariables method calls
```

File: core/Plugin/Controller.php [hunk 2]
```diff
@@ -809,6 +819,33 @@ protected function addCustomLogoInfo($view)
...
+    /**
+     * Set the template variables to show the what's new popup if appropriate
+     *
+     * @param View $view
+     * @return void
+     */
+    protected function showWhatIsNew(View $view): void
+    {
+        $view->whatisnewShow = false;
+        $view->whatisnewTooltip = addslashes(Piwik::translate('CoreAdminHome_WhatIsNewTooltip'));
```
Comment (sgiehl) [inline core/Plugin/Controller.php]: would be better to do the escaping the in view where needed. Actually the full translation could be done within the view, unless there is a reason to push it from PHP to the view.

File: plugins/CorePluginsAdmin/CorePluginsAdmin.php [hunk 3]
```diff
@@ -54,8 +54,18 @@ public function addPluginChanges(string $pluginName)
...
+    }
+
+    /**
+     * Retrieve an instantiated ChangesModel object
+     *
+     * @return ChangesModel
+     */
+    private function getChangesModel(): ChangesModel
+    {
+        $changesModelClass = StaticContainer::get(\Piwik\Changes\Model::class);
```
Comment (sgiehl) [inline plugins/CorePluginsAdmin/CorePluginsAdmin.php]: The StaticContainer should normally directly return a class instance, so there should be no need to instantiate it below. Parameters for the constructor should be provided automatically if they can be resolved. If not you might need to provide that in config.php

There are a couple of other places, where this is used similar and should be changed.

**Review by sgiehl — CHANGES_REQUESTED** [review]
Left a couple of comments. Besides that it might be good to add some UI tests for that badge count display.

File: core/Plugin/Controller.php (hunk 1 above)
Comment (bx80) [reply core/Plugin/Controller.php]: Fixed, need to work on my English as a first language :+1:

File: core/Plugin/Controller.php (hunk 2 above)
Comment (bx80) [reply core/Plugin/Controller.php]: I've moved the translation of the tooltip to the view, I couldn't find a suitable twig filter to add slashes for the apostrophe so I did this in the JS to keep it all in the template.

File: plugins/CorePluginsAdmin/CorePluginsAdmin.php (hunk 3 above)
Comment (bx80) [reply plugins/CorePluginsAdmin/CorePluginsAdmin.php]: Good point, I've now fixed this. The constructors to the model class were unnecessary so I've removed them.

**Pull request closed without merging**